### ISSUE TITLE
Soft hyphen (00AD) is invisible

### ISSUE THREAD

**Issue by drs-p** [canonical/Ubuntu-Sans-fonts#131]
The soft hyphen in the Ubuntu fonts is invisible and has width zero. That is a problem in a Tagged PDF as soft hyphens should be used for hyphenated words to improve accessibility and screen reading. See the discussion at this issue: https://github.com/latex3/tagging-project/issues/967, and also https://jkorpela.fi/shy.html#pdf for a discussion about the soft hyphen in general and its use in PDF specifically.

**Comment by djrrb** [comment]
I am not a decider on what happens next, but I figured I’d provide some context if helpful:

The latest version of this font was developed to pass fontbakery quality assurance tests. At the time, the [fontbakery recommendation](https://github.com/fonttools/fontbakery/issues/4046#issue-1568929553) was: 

> This font has a 'Soft Hyphen' character (codepoint 0x00AD) which is supposed to be zero-width and invisible, and is used to mark a hyphenation possibility within a word in the absence of or overriding dictionary hyphenation. It is mostly an obsolete mechanism now, and the character is only included in fonts for legacy codepage coverage. [code: softhyphen]

This is echoed in the [Unicode recommendation](https://www.unicode.org/reports/tr14/tr14-49.html#SoftHyphen):

> Unlike U+2010 HYPHEN, which always has a visible rendition, the character U+00AD SOFT HYPHEN (SHY) is an invisible format character that merely indicates a preferred intraword line break position. 

I see in your links that there has been much discussion about this on the PDF side, and there has also been [much](https://github.com/fonttools/fontbakery/issues/4095) [discussion](https://typedrawers.com/discussion/2046/special-dash-things-softhyphen-horizontalbar) about this on the font side as well.

The latest fontbakery recommendation is to [not include the character at all](https://github.com/fonttools/fontbakery/blob/main/Lib/fontbakery/checks/soft_hyphen.py).

So there is a good argument to removing this glyph from the next version of Ubuntu, but I think that fulfilling your request would cause the fonts to fail fontbakery tests.

**Comment by drs-p** [comment]
Thank you for your elaborate answer. There appears to be a mismatch between the font world and the tagged pdf world here. I don't know how the tagged pdf side should proceed, but I understand your decision not to adapt the font. I'll close this issue.

**Comment by djrrb** [comment]
I agree there seems to be a mismatch! 

The argument for the [fontbakery](https://github.com/fonttools/fontbakery) / [fontspector](https://github.com/fonttools/fontspector) interpretation is that it follows Unicode’s definition of the character. 

But if it presents a real-word accessibility issue for tagged PDFs, and there is no harm in giving the soft hyphen an outline and a positive width (as many fonts already do), then perhaps there is an argument to be made to the folks at fontbakery/fontspector that it is worth revising their recommendation.

If you want to pursue this, I think filing an issue with them would be a good way to continue the conversation. And if their recommendation changes, I think you’ll find that this and many other font projects would be happy to follow suit. 

(I’ll also add that, worst case scenario, it would be relatively simple to use [fonttools](https://github.com/fonttools/fonttools) to develop a tool that would “patch” the fonts to work as you expect.)
